evolver/app/routers/hardware.py

```diff
@@ -66,69 +69,86 @@ def hardware_commit(hardware_name: str, request: Request):
 
 
 # Start the calibration procedure for the selected hardware and vials
-@router.post("/{hardware_name}/calibrator/start")
+@router.post("/{hardware_name}/calibrator/procedure/start")
 def start_calibration_procedure(
-    hardware_name: str, request: Request, calibration_request: StartCalibrationProcedureRequest
+    hardware_name: str,
+    request: Request,
+    initial_state: Dict | None,
 ):
     hardware_instance = get_hardware_instance(request, hardware_name)
     calibrator = hardware_instance.calibrator
     if not calibrator:
-        raise HTTPException(status_code=404, detail=f"Calibrator not found for '{hardware_name}'")
-
-    calibrator.initialize_calibration_procedure(
-        selected_hardware=hardware_instance,
-        selected_vials=calibration_request.selected_vials,
-        evolver=request.app.state.evolver,
-    )
-
-    return calibrator.state
+        raise CalibratorNotFoundError
+    if hasattr(calibrator, "initialize_calibration_procedure"):
+        calibrator.initialize_calibration_procedure(
+            selected_hardware=hardware_instance,
+            initial_state=initial_state,
+        )
+    else:
+        raise CalibratorCalibrationProcedureFailedToInitializeError
```

Comment:
Ok I refactored (renamed) `initialize_calibration_procedure` to `create_calibration_procedure` and made it an abstract method so that now, this `hasattr` is not needed as you comment.

Reply:
fixed. got rid of the hasattr.